poseidon2/types.nim

```diff
@@ -18,9 +18,17 @@ func getZero*() : F =
   setZero(z)
   return z
 
+func getOne*() : F = 
+  var y : F
+  # y.fromUint(1'u32)       # WTF, why does this not compile ???
```

Comment:
The implementation of `fromUint` uses a cast from uint32 to array of byte, which isn't supported at compile time. The statement `const one*  : F = getOne()` in `roundfun.nim` is evaluated at compile time.

Reply:
Ah... And why is this simple casting not supported at compile time?

Reply:
I don't know